command_line/find_spots.py

```diff
@@ -135,11 +135,24 @@ def run(self, args=None):
         if not params.output.shoeboxes:
             del reflections["shoebox"]
 
-        # ascii spot count per image plot
+        # ascii spot count per image plot - per imageset
+
+        imagesets = []
+        imageset_experiment = {}
         for i, experiment in enumerate(experiments):
-            ascii_plot = spot_counts_per_image_plot(
-                reflections.select(reflections["id"] == i)
-            )
+            if experiment.imageset not in imagesets:
+                imagesets.append(experiment.imageset)
+                imageset_experiment[experiment.imageset] = [i]
+            else:
+                imageset_experiment[experiment.imageset].append(i)
+
```

Comment:
You create imageset_experiment but it doesn't appear to be used for anything. You could probably simplify these changes by doing the following . 

You probably don't need to maintain a list of imagesets separate to the dictionary of (imageset, experiment) pairs. Perhaps change this to something like:

```
from collections import defaultdict
imageset_experiment = defaultdict(list)
for i, experiment in enumerate(experiments):
    imageset_experiment[experiment.imageset].append(i)
```

Then instead of the lines below you could have:

```
        for imageset in imageset_experiment.keys():
            selected = flex.bool(reflections.nrows(), False)
            for i in imageset_experiment[imageset]:
                selected.set_selected(reflections["id"] == i, True)
                ascii_plot = spot_counts_per_image_plot(reflections.select(selected))
```

Reply:
I'm happy to make these changes

Reply:
Thanks for the suggestion - I'm happy to make the changes as I would like to fully understand everything which is happening here (bus # increment)

Reply:
Done 👍